src/main/java/org/carlmontrobotics/robotcode2023/subsystems/Arm.java

```diff
@@ -15,148 +15,98 @@
-        GoalPos.STORED[0].wristPos = SmartDashboard.getNumber("StoredWristCube", GoalPos.STORED[0].wristPos);
-        GoalPos.SHELF[0].armPos = SmartDashboard.getNumber("ShelfArmCube", GoalPos.SHELF[0].armPos);
-        GoalPos.SHELF[0].wristPos = SmartDashboard.getNumber("ShelfWristCube", GoalPos.SHELF[0].wristPos);
-        GoalPos.SUBSTATION[0].armPos = SmartDashboard.getNumber("SubstationArmCube", GoalPos.SUBSTATION[0].armPos);
-        GoalPos.SUBSTATION[0].wristPos = SmartDashboard.getNumber("SubstationWristCube", GoalPos.SUBSTATION[0].wristPos);
-        GoalPos.INTAKE[0].armPos = SmartDashboard.getNumber("IntakeArmCube", GoalPos.INTAKE[0].armPos);
-        GoalPos.INTAKE[0].wristPos = SmartDashboard.getNumber("IntakeWristCube", GoalPos.INTAKE[0].wristPos);
-
-        GoalPos.LOW[1].armPos = SmartDashboard.getNumber("LowArmCone", GoalPos.LOW[1].armPos);
-        GoalPos.LOW[1].wristPos = SmartDashboard.getNumber("LowWristCone", GoalPos.LOW[1].wristPos);
-        GoalPos.MID[1].armPos = SmartDashboard.getNumber("MidArmCone", GoalPos.MID[1].armPos);
-        GoalPos.MID[1].wristPos = SmartDashboard.getNumber("MidWristCone", GoalPos.MID[1].wristPos);
-        GoalPos.HIGH[1].armPos = SmartDashboard.getNumber("HighArmCone", GoalPos.HIGH[1].armPos);
-        GoalPos.HIGH[1].wristPos = SmartDashboard.getNumber("HighWristCone", GoalPos.HIGH[1].wristPos);
-        GoalPos.STORED[1].armPos = SmartDashboard.getNumber("StoredArmCone", GoalPos.STORED[1].armPos);
-        GoalPos.STORED[1].wristPos = SmartDashboard.getNumber("StoredWristCone", GoalPos.STORED[1].wristPos);
-        GoalPos.SHELF[1].armPos = SmartDashboard.getNumber("ShelfArmCone", GoalPos.SHELF[1].armPos);
-        GoalPos.SHELF[1].wristPos = SmartDashboard.getNumber("ShelfWristCone", GoalPos.SHELF[1].wristPos);
-        GoalPos.SUBSTATION[1].armPos = SmartDashboard.getNumber("SubstationArmCone", GoalPos.SUBSTATION[1].armPos);
-        GoalPos.SUBSTATION[1].wristPos = SmartDashboard.getNumber("SubstationWristCone", GoalPos.SUBSTATION[1].wristPos);
-        GoalPos.INTAKE[1].armPos = SmartDashboard.getNumber("IntakeArmCone", GoalPos.INTAKE[1].armPos);
-        GoalPos.INTAKE[1].wristPos = SmartDashboard.getNumber("IntakeWristCone", GoalPos.INTAKE[1].wristPos);
-
-        MAX_FF_VEL = SmartDashboard.getNumber("Max FF Vel", MAX_FF_VEL);
-        MAX_FF_ACCEL = SmartDashboard.getNumber("Max FF Accel", MAX_FF_ACCEL);
-    }
-
-    public void setArmVoltage(double volts) {
+        getArmControlOnSmartDashboard();
+    }
+ 
+    public void driveArm(double vel, double accel, boolean usePID) {
+        double armFeedVolts = getKg() * getCoM().getAngle().getCos() + armFeed.calculate(vel, accel);
+        double armPIDVolts = armPID.calculate(armEncoder.getPosition(), goalPosRad[ARM]);
+        if (getArmPos() > ARM_UPPER_LIMIT_RAD || getArmPos() < ARM_LOWER_LIMIT_RAD) {
+            armFeedVolts = getKg() * getCoM().getAngle().getCos() + armFeed.calculate(0, 0);
+        }
```

Comment:
This means if we ever go out of range that the arm will be held in position. Can you add a velocity check so that it moves if we try to move it back into range.